middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -590,33 +823,34 @@ let close_apply acc env ({ kind; func; args; continuation; exn_continuation;
   let apply =
     Apply.create ~callee
       ~continuation:(Return continuation)
-      exn_continuation
+      apply_exn_continuation
       ~args
       ~call_kind
       (Debuginfo.from_location loc)
-      ~inline:(LC.inline_attribute inlined)
+      ~inline:inline_call
       ~inlining_state:(Inlining_state.default ~round:0)
       ~probe_name
   in
-  Expr_with_acc.create_apply acc apply
+  match Inlining.inlinable env callee ~inline_call with
+  | Not_inlinable ->
+    Expr_with_acc.create_apply acc apply
+  | Inlinable func_desc ->
+    Inlining.inline acc ~apply ~apply_depth:(Env.current_depth env) ~func_desc
```

Comment:
You should add a call to `Inlining_report.record_decision` in each branch to record the decision (whether it be to inline or not to inline).